PSAlphaFS.psm1

```diff
@@ -552,43 +576,101 @@ function Copy-LongItem
+			#if( -not ( $DirObject::Exists($Destination)))
+			#{
+			#	$null = $DirObject::CreateDirectory($Destination)
+			#} 
+			#Else
+            #{
+			#	$Destination = $PathFSObject::Combine($Destination, $basename) 
+			#}
+            try {
+                $null = $DirObject::CreateDirectory($Destination)
+            }
+            catch [Alphaleonis.Win32.Filesystem.AlreadyExistsException]
 			{
-				$Destination = $PathFSObject::Combine($Destination, $basename)
+				
 			}
 		}#destination is a folder
         
 
-                
-		if($PathObject.EntryInfo.IsDirectory)
-		{
-			$fsObject = $DirObject
+		# Perform the copy     
+		
+		Write-Verbose -Message ("Copy-LongItem:`tCopying '{0}' to '{1}'" -f $Path, $Destination)
+        if($PathObject.EntryInfo.IsDirectory)
+		{			    
+            if ($isFile) {
+                Write-Warning -Message ("Copy-LongItem:`tCan not copy a directory '{0}' into a file {1}." -f $Path, $Destination)
+                return
+            }
+            else {
+                # Directory -> Directory
+                try {
+                    $DirObject::Copy($Path, $Destination, $copyOptions)  
+                }
+                catch [Alphaleonis.Win32.Filesystem.AlreadyExistsException] {
+                    # can get clever in here later and make it a mini robocop
+                    Write-Warning -Message ("Copy-LongItem:`tDirectory to Directory Copy with existing file conflicts not supported")                
+                    return
+                }
+            }
 		}
 		else
 		{
-			$fsObject = $FileObject
-		}    
+            try {
+                if ($isFile) {
+                    # File -> File
+                    $FileObject::Copy($Path, $Destination, $copyOptions)                
+                }
+                else {
+                    # File -> Directory
+                    $FileObject::Copy($Path, $PathFSObject::Combine($Destination, $basename), $copyOptions)
+                }    
+            }
+            catch [Alphaleonis.Win32.Filesystem.AlreadyExistsException] {
+                if($Force)
+			    {                
+				    # need to delete destination because the copy method dosent contain an overload that includes
+				    # both the overwrite and copyoptions parameters                    
+
+				    Write-Verbose -Message ("Copy-LongItem:`t Overwriting existing item...Copying '{0}' to '{1}'" -f $Path, $Destination)
+				    if ($isFile) {
+                        # File -> File
+                        $FileObject::Delete($Destination, $true, $PathFSFormatObject::FullPath)                    
+                        $FileObject::Copy($Path, $Destination, $copyOptions)                
+                    }
+                    else {
+                        # File -> Directory
+                        $FileObject::Delete($PathFSObject::Combine($Destination, $basename), $true, $PathFSFormatObject::FullPath)                    
```

Comment:
we need to handle the possibility of the existence of a folder with the same name as the file : 

                        if($DirObject::Exists($Destination))
                        {
                            Write-Warning -Message ("Copy-LongItem:`tCannot create a file at [{0}] when a directory already exists with the same name" -f $Destination) 
                            return
                        }